kong.conf.default

```diff
@@ -1653,6 +1653,32 @@
                                  #   This flavor of router will be removed in the next
                                  #   major release of Kong.
 
+max_request_headers = 100        # Maximum number of request headers to parse by default.
+                                 #
+                                 # This argument can be set to zero to remove the limit
+                                 # and to process all request headers. Removing the cap
+                                 # is strongly discouraged.
+
+max_response_headers = 100       # Maximum number of response headers to parse by default.
+                                 #
+                                 # This argument can be set to zero to remove the limit
+                                 # and to process all request headers. Removing the cap
+                                 # is strongly discouraged.
+
+max_uri_args = 100               # Maximum number of request uri arguments to parse by
+                                 # default.
+                                 #
+                                 # This argument can be set to zero to remove the limit
+                                 # and to process all request headers. Removing the cap
+                                 # is strongly discouraged.
+
+max_post_args = 100              # Maximum number of request post arguments to parse by
+                                 # default.
+                                 #
+                                 # This argument can be set to zero to remove the limit
+                                 # and to process all request headers. Removing the cap
+                                 # is strongly discouraged.
```

Comment:
Since there are significant downsides to removing the cap, should we even give user an option to remove the limit?

The user can choose to set the value to a very high value if needed. In my conversation https://github.com/Kong/kong/issues/10324 it seems like making this value configurable is sufficient.
Is there a good use-case for giving the power to remove the limit altogether?

Reply:
@hbagdi on PDK we have maximum `cap` or `1000` (and minumum of `1`). The `0` here can override the `1` minimum in PDK. Perhaps having this between 1 and 1000 is good enough? This doesn't apply to proxying though. The proxy just proxies everything, so these limitations apply mostly on Kong and Plugin code that tries to use the functions mentioned in this PR description (I add it there).

Reply:
> The 0 here can override the 1 minimum in PDK.

I'm confused. The docs in kong.conf.default says that the 0 removes the upper limit.

Which one is it?
I recommend that we put a maximum configurable limit of 1000 (even via kong.conf) and do not provide a knob to disable these limits.
I really can't think of use cases that have more than 1000 args, I mean someone can do it but that's clearly bad and we should encourage the user to fix the API rather than making it easier to build such APIs.

What do you think?

Reply:
@hbagdi, No need to be confused. kong.conf 0 removes the boudary both ngx.* and kong..* but if you try to give 0 explicitly to kong.* it will error out (it has allats had this limit).